lib/spack/spack/modules/common.py

```diff
@@ -741,6 +737,24 @@ def environment_modifications(self):
             exclude=spack.util.environment.is_system_path
         )
 
+        # BlueBrain: we do not generate modules for dependencies, include
+        # their modifications and bin/ directories here
+        #
+        # This should be done here to let the current package prepend paths
+        # properly.
+        #
+        # First do some setup for all dependencies, then execute
+        # modifications
+        for dep in set(spec.traverse(root=False, deptype='run')):
+            dpkg = dep.package
+            dpkg.setup_dependent_package(spec.package.module, spec)
+        for dep in set(spec.traverse(root=False, deptype='run')):
+            if not (dep.external and dep.external_modules):
+                dpkg = dep.package
```

Comment:
`dpkg` seems unnecessary, same w/ above

Reply:
yeah, trimmed that. hopefully passes the pedantic lint.